src/wp-includes/theme.php

```diff
@@ -4181,7 +4181,11 @@ function _add_default_theme_supports() {
 	add_theme_support( 'post-thumbnails' );
 	add_theme_support( 'responsive-embeds' );
 	add_theme_support( 'editor-styles' );
-	add_theme_support( 'html5', array( 'comment-form', 'comment-list', 'style', 'script' ) );
+	/*
+	 * Make block themes support HTML5 by default for all non-block content.
+	 * Blocks contain their own markup, which is HTML5.
```

Comment:
The comment is slightly incorrect as the comment blocks are using the default template functions. That's also why they are part of the current list and nothing else beside script/style.

Since block themes won't use any of the other template function I still think that none of the other features should be added as they are now kind of deprecated and provide no benefit.

Reply:
@ocean90 When a user upgrades to 5.9 with non-block content such as a gallery or image with a caption, the content uses the `[gallery]` and `[caption]` shortcodes. Without these HTML5 theme supports, these shortcodes will not use HTML5 markup.

### For example:

An original post non-block post that becomes this when upgrading to 5.9:
```html
<!-- wp:paragraph -->
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

&nbsp;

[gallery ids="12,13,14,15"]

&nbsp;

Now try with a picture and caption.

[caption id="attachment_14" align="alignnone" width="1023"]<img class="size-full wp-image-14" src="fall3.jpg" alt="" width="1023" height="685"> Autumn colorful foliage over lake with beautiful woods in red and yellow color.[/caption]</p>
<!-- /wp:paragraph -->
```

#### The `[gallery]` renders like this:

<img width="2480" alt="gallery -before-after" src="https://user-images.githubusercontent.com/7284611/148067685-9bc69697-1028-426e-a430-d590677999a2.png">

* Without HTML5 theme support: uses `dl`, `dt`, and `dd` with inline style tag.
* With HTML5 theme support: uses figure, div, and figcaption and no inline style tag.

#### The `[caption]` renders like this:

<img width="2483" alt="caption -before-after" src="https://user-images.githubusercontent.com/7284611/148068149-d78f7aef-9621-4c10-9693-4d63398d44ac.png">

* Without HTML5 theme support: uses `div` with `p` and an extra 10px of width.
* With HTML5 theme support: uses `figure` and `figcaption`.